sunpy/image/slit.py

```diff
@@ -0,0 +1,129 @@
+import sunpy
+import sunpy.map
+import sunpy.cm
+import sunpy.wcs
+
+import numpy as np
+import astropy.wcs
+import astropy.io.fits as fits
+import astropy.units as u
+
+import scipy.ndimage as snd
+import matplotlib.pyplot as plt
+from glob import glob
+
+
+
+def slit(in_files, xy1, xy2):
+    """
+    Returns an array with intensity along the slit on the y axis and time 
+    along x.
+    
+    Parameters    
+    ----------
+    in_files : `sunpy.map.MapCube`
+        A mapcube of the images you want to perform the slit analysis on. 
+        Usually with x and y as space, and z as time.
+    xy1 : List of x and y coordinates in `[x1, y1]`
+        The x and y coordinates of the beginning of the slit. May be either 
+        pixel coordinates or astropy units.
+    xy2 : List of x and y coordinates
+        The x and y coordinates of the end of the slit. May be either 
+        pixel coordinates or astropy units.
+
+        
+    Returns
+    -------
+    out : numpy array
+        A numpy array with intenisty in y and time in x.
+
+    """
+    
```

Comment:
The first check you should do is that all the Maps in the MapCube are the same shape.

(This is needed even for the data coordinates because you are only doing pixel -> data once, and you are assuming you will have a uniform size array out.